ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/models/Type.java

```diff
@@ -51,4 +49,26 @@ public String getUnqualifiedName() {
     public String getNameUnderscores() {
         return helper.camelToUnderscore(getUnqualifiedName());
     }
+
+    public String generateMatcher() {
+        final StringBuilder outputBuilder = new StringBuilder();
+        final int numConstants = this.enumConstants.size();
+        for (int currentIndex = 0; currentIndex < numConstants; currentIndex++) {
```

Comment:
Why not use a foreach `for (final Ds3EnumConstant enumConstant : this.enumConstants)`?